### ISSUE TITLE
On M1, CPU reported as Cortex a57 and many CPU features disabled

### ISSUE THREAD

**Issue by a74nh** [docker/for-mac#5681]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
Running lscpu on a fedora/ubuntu/otherlinux environment should show all the AArch64 flags available on the M1

```
$ lscpu
Architecture:                    aarch64
CPU op-mode(s):                  64-bit
Byte Order:                      Little Endian
CPU(s):                          4
On-line CPU(s) list:             0-3
Thread(s) per core:              1
Core(s) per socket:              4
Socket(s):                       1
Vendor ID:                       0x00
Model:                           0
Stepping:                        0x0
BogoMIPS:                        48.00
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma
                                 lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp f
                                 lagm2 frint
```

### Actual behavior

CPU is reported as a cortex a57, is missing many of the CPU flags such as atomics and paca, and the vendor ID is incorrect.

```
$ lscpu
Architecture:                    aarch64
CPU op-mode(s):                  64-bit
Byte Order:                      Little Endian
CPU(s):                          4
On-line CPU(s) list:             0-3
Thread(s) per core:              1
Core(s) per socket:              4
Socket(s):                       1
Vendor ID:                       ARM
Model:                           0
Model name:                      Cortex-A57
Stepping:                        r1p0
BogoMIPS:                        48.00
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm aes pmull sha1 sha2 crc32 fphp
                                  asimdhp cpuid dit
```

This is useful to me as I would like to use many of these features in my apps I'm running inside docker.

### Information

  Big Sur 11.3.1
  Macbook Pro M1
  Docker 3.3.3

  If I switch to "Use new virtualization framework" then the CPU is reported correctly. However, this is not useful as using this kills my VPN connections.

I suspect the issue may be down to docker using "-cpu cortex-a57" when running qemu:

```
$ ps -ef | grep qemu
  501  4781  4770   0  8:55am ??         2:00.32 /Applications/Docker.app/Contents/MacOS/qemu-system-aarch64 -accel hvf -cpu cortex-a57 -machine virt,highmem=off -m 2048 -smp 4 -kernel /Applications/Docker.app/Contents/Resources/linuxkit/kernel -append page_poison=1 vsyscall=emulate panic=1 nospec_store_bypass_disable noibrs noibpb no_stf_barrier mitigations=off   vpnkit.connect=tcp+bootstrap+client://192.168.65.2:52423/9bd6d886689f63985c301e170f7328100032dbd65d6d78113c68da5e5e638109 console=ttyAMA0 -initrd /Applications/Docker.app/Contents/Resources/linuxkit/initrd.img -serial pipe:/var/folders/_y/625pd_h11rdfg4gjdhj9fgcw0000gn/T/qemu-console709681769/fifo -drive if=none,file=/Users/alahay01/Library/Containers/com.docker.docker/Data/vms/0/data/Docker.raw,format=raw,id=hd0 -device virtio-blk-pci,drive=hd0,serial=dummyserial -netdev socket,id=net1,fd=3 -device virtio-net-device,netdev=net1,mac=02:50:00:00:00:01 -vga none -nographic -monitor none
```

### Steps to reproduce the behavior

 
  1. docker run --name fedora_test -t fedora
  2. docker exec -it fedora_test lscpu

**Comment by stephen-turner** [comment]
Since 3.3.2, we use `-cpu max` when running qemu, our understanding was that that should include everything available.

**Comment by a74nh** [comment]
> Since 3.3.2, we use `-cpu max` when running qemu, our understanding was that that should include everything available.

Using cpu max would fix the issue.

But, looking at my ps output (pasted above) it is still using "-cpu cortex-a57".

It's possible that my 3.3.3 install is using setting from an older version? I have clicked both "clean/purge data" and "reset to factory defaults", but this didn't change anything.

**Comment by stephen-turner** [comment]
I don't think so, and I also don't think we ever used cortex-a57 explicitly. Let me consult.

**Comment by stephen-turner** [comment]
So I was wrong. There are two qemu's in play here, an "outer" one acting as a hypervisor for running a Linux VM, and an "inner" one acting as an emulator for running amd64 containers inside the VM. We set the inner one to `-cpu max` but not the outer one. We're now trying `-cpu host` on the outer one.

**Comment by djs55** [comment]
@a74nh I've made a test build using `-cpu host`: https://desktop-stage.docker.com/mac/stable/arm64/64397/Docker.dmg . Note that it hasn't been notarised.

I see the following:
```
% docker run -it ubuntu bash
Unable to find image 'ubuntu:latest' locally
latest: Pulling from library/ubuntu
80bc30679ac1: Pull complete 
9bf18fab4cfb: Pull complete 
5979309c983a: Pull complete 
Digest: sha256:cf31af331f38d1d7158470e095b132acd126a7180a54f263d386da88eb681d93
Status: Downloaded newer image for ubuntu:latest
root@4d8e2fd0b9f9:/# lscpu
Architecture:                    aarch64
CPU op-mode(s):                  64-bit
Byte Order:                      Little Endian
CPU(s):                          4
On-line CPU(s) list:             0-3
Thread(s) per core:              1
Core(s) per socket:              4
Socket(s):                       1
Vendor ID:                       0x00
Model:                           0
Stepping:                        0x0
BogoMIPS:                        48.00
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 asimddp sha512 asimdfhm dit uscat ilrcpc flagm sb paca pacg dcpodp flagm2 frint
```

This build should also have a fix for using VPNs with `virtualization.framework` -- let me know if that works for you. If not, check that the VPN-safe networking has been enabled by editing `~/Library/Group\ Containers/group.com.docker/settings.json` and checking that it has `"useVpnkit": true,`.

**Comment by a74nh** [comment]
That new image fixes both my issues:
* standard virtualisation is using cpu=host, and I can see all the CPU features I wanted.
* New virtualization framework works with my VPN.

Thanks for looking into these!

**Comment by djs55** [comment]
@a74nh thanks for the quick feedback, I'm glad the fixes worked for you!

**Comment by mat007** [comment]
A fix has been released in Docker Desktop 3.4.0, see [the release notes](https://docs.docker.com/docker-for-windows/release-notes/#bug-fixes-and-minor-changes)!
